extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/DevServicesLambdaProcessor.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.amazon.lambda.deployment;
+
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.amazon.lambda.runtime.AmazonLambdaApi;
+import io.quarkus.amazon.lambda.runtime.LambdaHotReplacementRecorder;
+import io.quarkus.amazon.lambda.runtime.MockEventServer;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.DevServicesNativeConfigResultBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.runtime.LaunchMode;
+
+public class DevServicesLambdaProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesLambdaProcessor.class);
+
+    static MockEventServer server;
+    static boolean first = true;
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    @Record(STATIC_INIT)
+    public void enableHotReplacementChecker(LaunchModeBuildItem launchMode,
+            LambdaHotReplacementRecorder recorder,
+            LambdaObjectMapperInitializedBuildItem dependency) {
+        if (launchMode.getLaunchMode().isDevOrTest()) {
+            try {
+                // if deprecated test framework in classpath, do not enable
+                Class.forName("io.quarkus.amazon.lambda.test.LambdaClient");
+            } catch (ClassNotFoundException e) {
+                recorder.enable();
+            }
+        }
+    }
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    public void startEventServer(LaunchModeBuildItem launchMode,
+            Optional<EventServerOverrideBuildItem> override,
+            BuildProducer<DevServicesNativeConfigResultBuildItem> devServicePropertiesProducer,
+            BuildProducer<ServiceStartBuildItem> serviceStartBuildItemBuildProducer) throws Exception {
+        if (!launchMode.getLaunchMode().isDevOrTest())
+            return;
+
+        if (server != null) {
+            if (launchMode.getLaunchMode() == LaunchMode.TEST) {
```

Comment:
For test mode this should just be shutdown when the current context ClassLoader is closed. Otherwise the server will hang around after the next run is done.

I should really add a nicer API for this, it's something that we are using a lot more now we have continuous testing and dev services.

Reply:
Actually current ClassLoader is not right either, I will add something for this.